### ISSUE TITLE
[Feature Request] Apply reset zoom to all tabs 

### ISSUE THREAD

**Issue by eizanprime** [stefanvd/Browser-Extensions#3]
As HiDPI support is still nowhere on linux, I am using this extension to have different zoom levels for different screen resolutions. I think a lot of other users use this extension for the same reason.

However, when I switch screens I need to refresh the tabs in order to adapt the zoom, which does take a lot of time, so it would be very practical if there was a button that would allow us to do it. Even better, do it automatically when there is a resolution change detected. 

Thank you very much for the very useful extension !

**Comment by stefanvd** [comment]
Thank you for your feedback!
For simplicity in the user experience, automatically solution will come in the next Zoom browser extension version.

**Comment by eizanprime** [comment]
Thank you for the reply !
I am looking forward to the next version ! Do you know when you are planning to release it ? 
Also it would be super practical if you could also add default zoom level per screen and domain ! (for example all sites are zoom level 100 in 1k srceen, 200 on 4k srceen, but just outlook is at 150 for 1k screen, and 300 for 4k screen ) 

Thank you very much !

**Comment by stefanvd** [comment]
The update is submitted to the Chrome web store (and is pending on Google side). Zoom Chrome extension version 2.5.4.0 has in the  background page this auto refresh:
// update when screen changed
chrome.system.display.onDisplayChanged.addListener(function(){
    chrome.tabs.query({},
        function(tabs){
            tabs.forEach(function(tab){
                chrome.tabs.sendMessage(tab.id,{text:'refreshscreen'});
            });
    });
});

**Comment by eizanprime** [comment]
Thank you so much ! I didn't know google took so much time to process this kind of update.. I am looking forward to it ! 

Btw are you planning to update this repo ?

**Comment by stefanvd** [comment]
Yes. I am working to get my extension folder on my new hard disk. And then I can update the complete Stefan vd Browser extension project.

**Comment by eizanprime** [comment]
The update finally arrived in the store ! However it seemingly does not work (nothing changed, I must still manually press the reset button for each tab when I change screens)
Is there any way I can maybe make it work ? 
Thank you very much !

**Comment by stefanvd** [comment]
That is strange. When you received the Zoom Chrome extension update, have you tried to restart your Chrome web browser. That to make sure it uses the new 'content.js' and 'background.js' script.

What if you try to change your screen with the Windows settings (on the current computer screen), do you see it also change the zoom value?

**Comment by photon3108** [comment]
I really need this feature, too. But my extension gets some problems. Before about 1~2 days ago, the extension can zoom correctly according to multiple screen size setting of extension. But it's not now. I have three monitors.
**Built-in Monitor** is 2560x1440 and the scale is 1.5. `Zoom` detect it as 1600x900 and I set it zoom to 125%. 
**External Monitor 1** is 3840x2160 and the scale is 1.62. `Zoom` detect it as 2400x1350 and I set it zoom to 150%. 
**External Monitor 2** is 2160x3840 and the scale is 1.5. `Zoom` detect it as 1350x2400 and I set it zoom to 110%.

Page zoom of `chrome://settings` is 175% (only for test) and `Use the built-in browser zoom` of `zoom` setting is enabled. `For all screen sizes` and `Apply the zoom level to all pages` of `zoom` setting are both disabled. When I refresh [yahoo](http://us.yahoo.com) page on **external monitor 1**, firstly it was zoomed to 175%, and then it was zoomed to 125%.

Is it possible to push a special version which is able to log what happens to chrome web store? And I could post log file here.

OS: Ubuntu 18.04 (Unity)
Zoom: 2.5.4.0

**Comment by eizanprime** [comment]
It looks like quite a few things are broken.. It tries to force a 200% zoom no matter the screen in my case.. I never used javasrcipt but I will try to modify a few things in the extension code..

**Comment by eizanprime** [comment]
What seems to be happening is that anytime I refresh a page or open a new one, the zoom level is set at the highest value (and not at the appropriate one for the screen), it breaks everything quite completely :/

**Comment by jonaszuberbuehler** [comment]
I'm experiencing the same bug as @eizanprime.

After digging through the code I think setting `chromedisplay` to the first returned screen info is causing this:

```
chromedisplay = screens[0].bounds.width + "x" + screens[0].bounds.height;
```

which gets later picked up by 

```js
allzoomvalue = screenzoom[satbbuf[i]] / 100;
```

Could you instead use `request.screen` to get the resolution of the current screen @stefanvd? 

Disclaimer: I have no experience at all with developing chrome extensions.

**Comment by eizanprime** [comment]
@jonaszuberbuehler TBH I abandonned using this extension, I made my own extention that updates all the zoom in function of the resolution when pressing the button

**Comment by stefanvd** [comment]
Thank you for your research! I am working on updating this to the Search Results chrome.system.display.

**Comment by jonaszuberbuehler** [comment]
Cheers!

Would have created a PR but the source code here seems to be outdated.

**Comment by stefanvd** [comment]
@jonaszuberbuehler I updated all my browser extensions here on Github to the latest public version (that include also the Microsoft Edge extensions).

**Comment by stefanvd** [comment]
@eizanprime 
@jonaszuberbuehler 
Fixed the issue in the Zoom Browser extension version 2.5.7.0. I add a new right-click menu item when you click right on the Z icon. You can then click on this menu item "Apply reset zoom value to all tabs". And that will reset all the zoom value of each tab, and apply the correct zoom value for that screen size. The "chrome.system.display" is removed, because this does not detect the automatic screen size change. It only detects when you connect a new monitor.

The update is currently pending the Chrome web store and Microsoft Edge store.

**Comment by jonaszuberbuehler** [comment]
Thx! Will give it a try as soon as it's approved.
